components/style/properties/longhand/font.mako.rs

```diff
@@ -1006,3 +1006,865 @@ ${helpers.single_keyword("font-variant-position",
+                        }
+
+                        ruby = true;
+                    },
+                    "jis78" => {
+                        if variant.is_some() {
+                            return Err(());
+                        }
+
+                        variant = Some(VariantValue::Jis78);
+                    },
+                    "jis83" => {
+                        if variant.is_some() {
+                            return Err(());
+                        }
+
+                        variant = Some(VariantValue::Jis83);
+                    },
+                    "jis90" => {
+                        if variant.is_some() {
+                            return Err(());
+                        }
+
+                        variant = Some(VariantValue::Jis90);
+                    },
+                    "simplified" => {
+                        if variant.is_some() {
+                            return Err(());
+                        }
+
+                        variant = Some(VariantValue::Simplified);
+                    },
+                    "traditional" => {
+                        if variant.is_some() {
+                            return Err(());
+                        }
+
+                        variant = Some(VariantValue::Traditional);
+                    },
+                    "full-width" => {
+                        if width.is_some() {
+                            return Err(());
+                        }
+
+                        width = Some(WidthValue::Full);
+                    },
+                    "proportional-width" => {
+                        if width.is_some() {
+                            return Err(());
+                        }
+
+                        width = Some(WidthValue::Proportional);
+                    },
+                    _ => break,
+                }
+            } else {
+                break;
+            }
+        }
+
+        Ok(SpecifiedValue::Style(Style{
+            ruby: ruby,
+            variant: variant,
+            width: width,
+        }))
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::Normal
+    }
+
+    #[inline]
+    pub fn get_initial_specified_value() -> SpecifiedValue {
+        SpecifiedValue::Normal
+    }
+</%helpers:longhand>
+
+<%helpers:longhand name="font-variant-ligatures" products="none" 
+                   animatable="False" spec="https://drafts.csswg.org/css-fonts-3/#font-variant-ligatures-prop">
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::HasViewportPercentage;
+    use values::computed::ComputedValueAsSpecified;
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    no_viewport_percentage!(SpecifiedValue);
+
+    pub mod computed_value {
+        pub use super::SpecifiedValue as T;
+    }
+
+    
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct Style {
```

Comment:
I think this could be written with `bitflags!` (a group of flags for representing the value itself, and the other group for representing whether the values have `non-` prefix). I'll leave it to @emilio though :)